File: src/main/java/it/discovery/model/Genre.java
package it.discovery.model;

import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@ToString
@EqualsAndHashCode(of = "name")
//@AllArgsConstructor
public class Genre {

    private String name;

    private LocalDateTime created;

    public Genre(String name, LocalDateTime created) {
        this.name = name;
        this.created = created;
    }
}

File: src/main/kotlin/it/discovery/kotlin/model/Genre.kt
package it.discovery.kotlin.model

import java.time.LocalDateTime

data class Genre(
    private val name: String, private
    val created: LocalDateTime
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        return other is Genre && name == other.name
    }

    override fun hashCode()  = name.hashCode()
}
